src/test/java/org/mule/extension/sftp/PostActionGroupTestCase.java

```diff
@@ -40,4 +63,194 @@ public void deleteAndMove() {
   public void deleteAndRename() {
     new PostActionGroup(true, null, "thisone.txt", true).validateSelf();
   }
+
+
+  @Test(expected = FileAlreadyExistsException.class)
+  @Description("verifies that move file with overwrite flag set to false and exceptiopn will be thrown")
+  public void testMoveWithoutOverwriteWhenFileExists() {
+    FileAttributes fileAttributes = mock(FileAttributes.class);
+    when(fileAttributes.getName()).thenReturn("");
+    FileConnectorConfig fileConnectorConfig = mock(FileConnectorConfig.class);
+
+    ConcreteFileSystem fileSystem = new ConcreteFileSystem("");
+
+    new PostActionGroupChild("someDir", null, false, false).apply(fileSystem, fileAttributes, fileConnectorConfig);
```

Comment:
I recommend you finish with a fail sentence (if the exception throws ok, the fail sentence will never be reached executed

Reply:
Like this:
```
@Test(expected = FileAlreadyExistsException.class)
public void expectedException() {
        methodThrowsException();
        fail("FileAlreadyExistsException exception was not thrown");
}
```

Reply:
OK, got it.